### ISSUE TITLE
OCS user creation: bad request when password policy not passed

### ISSUE THREAD

**Issue by DevNIX** [nextcloud/server#7946]
After a good amount of time scratching my head, I think I found a bug (or, at least, a very misinformative error):

When I try to create a new user using the OCS API and the password does not meet the system's password policy, the server will return a "Bad request" message:

```
{"ocs"=>{"meta"=>{"status"=>"failure", "statuscode"=>101, "message"=>"Bad request", "totalitems"=>"", "itemsperpage"=>""}, "data"=>[]}}
```

**Comment by MorrisJobke** [comment]
Correct. I found the issue and will fix it.

**Comment by MorrisJobke** [comment]
Fix is in #8502

**Comment by MorrisJobke** [comment]
It's very likely that this will be also fixed in 12.0.6 and 13.0.1 ;)
